packages/angular_devkit/core/src/workspace/definitions.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonValue } from '../json';
+
+export interface WorkspaceDefinition {
+  readonly extensions: Record<string, JsonValue | undefined>;
+
+  readonly projects: ProjectDefinitionCollection;
+}
+
+export interface ProjectDefinition {
+  readonly extensions: Record<string, JsonValue | undefined>;
+  readonly targets: TargetDefinitionCollection;
+
+  root: string;
+  prefix?: string;
+  sourceRoot?: string;
+}
+
+export interface TargetDefinition {
+  options?: Record<string, JsonValue | undefined>;
+  configurations?: Record<string, Record<string, JsonValue | undefined> | undefined>;
+
+  builder: string;
+}
+
+export type DefinitionCollectionListener<V> = (
```

Comment:
What are our usecases for the listeners?

Reply:
Formats may require change tracking of the collections to facilitate fine-grained updates of the underlying storage medium.  The JSON format, for instance, allows for this.

Reply:
SGTM